napari/layers/_pyramid_layer/model.py

```diff
@@ -138,16 +153,17 @@ def get_value(self):
             shape = self._image_view.shape[:-1]
         else:
             shape = self._image_view.shape
+
         coord[-2:] = np.clip(coord[-2:], 0, np.subtract(shape, 1))
         value = self._image_view[tuple(coord[-2:])]
 
         pos_in_slice = (self.coordinates[-2:]
                         + self.translate[[1, 0]] / self.scale[:2])
 
         # Make sure pos in slice doesn't go off edge
-        shape = self._image_shapes[self._pyramid_level]
-        coord = np.clip(pos_in_slice, 0, np.subtract(shape, 1))
-        coord = np.round(coord * self.scale[:2]).astype(int)
+        shape = self._image_shapes[self.pyramid_level][-2:]
```

Comment:
Can you add a note above this line that this needs to change with the dims model?

Reply:
I have added the following note where it is needed
`# TODO: Change dims selection when dims model changes`